app/cdap/components/PipelineDiff/store/diffSlice.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { createSlice } from '@reduxjs/toolkit';
+import { MyPipelineApi } from 'api/pipeline';
+import { Observable } from 'rxjs/Observable';
+import { computePipelineDiff, getReactflowPipelineGraph } from './pipelineUtil';
+
+const initialState = {
+  isLoading: false,
+
+  // TODO: type pipelines
```

Comment:
why not type it in this PR?

Reply:
I do not know for certain what the types for these pipelines will be until I implement the diff algorithm and the new canvas component. I will type these as soon as I know enough about their values